app/controllers/foreman_inventory_upload/tasks_controller.rb

```diff
@@ -13,12 +13,21 @@ def create
           message: N_('Nothing to sync, there are no hosts with subscription for this organization.'),
         }, status: :method_not_allowed
       else
-        host_statuses = InventorySync::Async::InventoryFullSync.perform_now(selected_org)
+        task = ForemanTasks.async_task(InventorySync::Async::InventoryFullSync, selected_org)
       end
 
       render json: {
-        syncHosts: host_statuses[:sync],
-        disconnectHosts: host_statuses[:disconnect],
+        task: task,
+      }, status: :ok
+    end
+
+    def index
+      task = ForemanTasks::Task.where(label: 'InventorySync::Async::InventoryFullSync')
```

Comment:
`task` can be nil, if it was never executed.
It will fail later in the `render` statement

Reply:
I think if it's `nil` we can return status `404 :not_found` to the client
and we hide the chart/show some empty state for "run inventory sync"